File: LLD/InterviewExamples/Blinkit/Java/src/models/Order.java
package models;

import java.util.List;
import java.util.UUID;

public class Order {

    private String id;
    private List<OrderItem> orderItemList;
    private OrderStatus status;

    public Order(List<OrderItem> orderItemList) {
        id = UUID.randomUUID().toString();
        this.orderItemList = orderItemList;
        this.status = OrderStatus.ACCEPTED;
    }

    public String getId() {
        return this.id;
    }

    public List<OrderItem> getItemList() {
        return this.orderItemList;
    }

    public OrderStatus getStatus() {
        return this.status;
    }

    public OrderItem getOrderItemByProductName(String productName) {
        for(OrderItem item : orderItemList) {
            if(item.getProduct().getName().equals(productName)) {
                return item;
            }
        }

        return null;
    }
    public void setOrderStatus(OrderStatus status) {
        this.status = status;
    }
}

File: LLD/InterviewExamples/Blinkit/Node/src/models/order.js
const { randomUUID } = require('crypto')
const OrderItem = require('.orderItem');

class Order {

    #id
    #orderItemList
    #orderStatus
    constructor(orderItemList) {
        this.#id = randomUUID();
        this.#orderItemList = orderItemList;
        this.#orderStatus = "ACCEPTED"
    }

    getId() {
        return this.#id;
    }

    getOrderItemList() {
        //ideally should be created a deep copy
        return this.#orderItemList;
    }

    removeItem(itemName) {
        this.#orderItemList = this.#orderItemList.filter((item) => item.getProduct().getName().equals(itemName))
        console.log("Item removed");
    }

    updateItem(itemName, quantity) {
        
        for(let item of this.#orderItemList) {
            if(item.getProduct().getName().equals(itemName)) {
                item.setQuantity(quantity)
                break;
            }
        }
    }

    getOrderStatus() {
        return this.#orderStatus
    }

    updateOrderStatus(status) {
        this.#orderStatus = status
    }
}

module.exports = Order;